openvdb/python/CMakeLists.txt

```diff
@@ -1,49 +1,167 @@
+  INCLUDE_DIRECTORIES ( ../../ )
+ENDIF ()
+
+FIND_PACKAGE ( IlmBase REQUIRED COMPONENTS Half )
+FIND_PACKAGE ( TBB REQUIRED COMPONENTS tbb )
+FIND_PACKAGE ( ZLIB REQUIRED )
+
+IF ( USE_LOG4CPLUS )
+  FIND_PACKAGE ( Log4cplus REQUIRED )
+ENDIF ()
+
+IF ( USE_BLOSC )
+  FIND_PACKAGE ( Blosc ${MINIMUM_BLOSC_VERSION} REQUIRED )
+ENDIF ()
+
+# NumPy requires CMake 3.14
+IF ( USE_NUMPY AND ( ${CMAKE_VERSION} VERSION_LESS 3.14 ) )
+  MESSAGE ( FATAL_ERROR "No CMake support for FindPackage ( NumPy ) currently "
+    "available for OpenVDB. Requires CMake 3.14." )
+ENDIF ()
+
+SET ( OPENVDB_PYTHON_DEPS )
+
+# If CMake is < 3.12, use the old style python search and alias the variables
+# Note that the Interpreter component is only required for the python test
+IF ( ${CMAKE_VERSION} VERSION_LESS 3.12 )
+  FIND_PACKAGE ( PythonInterp ${PYTHON_VERSION} REQUIRED )
+  FIND_PACKAGE ( PythonLibs ${PYTHON_VERSION} REQUIRED )
+
+  SET ( Python_VERSION_MAJOR ${PYTHON_VERSION_MAJOR} )
+  SET ( Python_VERSION_MINOR ${PYTHON_VERSION_MINOR} )
+  SET ( Python_EXECUTABLE ${PYTHON_EXECUTABLE} )
+  SET ( OPENVDB_PYTHON_DEPS ${PYTHON_LIBRARIES} )
+  GET_FILENAME_COMPONENT ( Python_LIBRARY_DIRS ${OPENVDB_PYTHON_DEPS} DIRECTORY )
+
+  INCLUDE_DIRECTORIES ( SYSTEM ${PYTHON_INCLUDE_DIR} )
+ELSE ()
+  # To ensure consistent versions between components Interpreter, Compiler,
+  # Development and NumPy, specify all components at the same time when using
+  # FindPython
+  IF ( USE_NUMPY )
+    FIND_PACKAGE ( Python ${PYTHON_VERSION} REQUIRED COMPONENTS Interpreter Development NumPy )
+    LIST ( APPEND OPENVDB_PYTHON_DEPS Python::NumPy )
+  ELSE ()
+    FIND_PACKAGE ( Python ${PYTHON_VERSION} REQUIRED COMPONENTS Interpreter Development )
+  ENDIF ()
+  LIST ( APPEND OPENVDB_PYTHON_DEPS Python::Python )
+ENDIF ()
 
-FIND_PACKAGE ( PythonInterp REQUIRED )
-FIND_PACKAGE ( PythonLibs REQUIRED )
-FIND_PACKAGE ( Boost ${MINIMUM_BOOST_VERSION} REQUIRED COMPONENTS python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR} )
+IF ( BOOST_PYTHON_NAMESPACE_VERSIONING )
+  SET ( BOOST_PYTHON_LIB "python${Python_VERSION_MAJOR}${Python_VERSION_MINOR}" )
+ELSE ()
+  SET ( BOOST_PYTHON_LIB "python" )
+ENDIF ()
 
+FIND_PACKAGE ( Boost ${MINIMUM_BOOST_VERSION}
+  REQUIRED COMPONENTS iostreams system ${BOOST_PYTHON_LIB}
+)
 
-IF ( NOT OPENVDB_BUILD_CORE )
-  FIND_PACKAGE ( OpenVDB REQUIRED )
+IF ( UNIX )
+  FIND_PACKAGE ( Threads REQUIRED )
+ENDIF ()
 
-  FIND_PACKAGE ( ILMBase REQUIRED )
-  FIND_PACKAGE ( OpenEXR REQUIRED )
-  FIND_PACKAGE ( Blosc REQUIRED )
-  FIND_PACKAGE ( TBB REQUIRED )
-  FIND_PACKAGE ( ZLIB REQUIRED )
+# Set core deps. Note that the order here is important. If we're building against
+# Houdini 17.5 we must include OpenEXR and IlmBase deps first to ensure the normal
+# namespaced headers are used over the Houdini 17.5 ones (which will be imported
+# through targets like tbb which are still deployed with Houdini)
 
+SET ( OPENVDB_PYTHON_DEPENDANT_LIBS
```

Comment:
Spelling should always be "dependent" when used as an adjective.

Reply:
Done